vis.c

```diff
@@ -1590,8 +1589,7 @@ Regex *vis_regex(Vis *vis, const char *pattern) {
 		text_regex_free(regex);
 		return NULL;
 	}
-	if (len == 0)
-		register_put0(vis, &vis->registers[VIS_REG_SEARCH], pattern);
+	register_put(vis, &vis->registers[VIS_REG_SEARCH], pattern, len+1);
```

Comment:
The `+1` here is not very intuitive. For some reason, `register_get` gives us a length without the terminating null byte (which is still part of the buffer), but `register_put` expects a length including the null.